### PR TITLE
Add documentation for compiling on MacOS

### PR DESCRIPTION
**Pull request by dkao1978**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,9 +57,11 @@ fn hello(py: Python) -> PyResult<()> {
 Example library with python bindings:
 
 The following two files will build with `cargo build`, and will generate a python-compatible library.
-On macOS, you will need to rename the output from \*.dylib to \*.so.
+For MacOS, "-C link-arg=-undefined -C link-arg=dynamic_lookup" is required to build the library. Setup.py includes this by default. See examples/word-count.
```
Comment (messense) [inline README.md]: `setuptools-rust` not `Setup.py`

Comment (messense) [reply README.md]: Use `[examples/word-count](examples/word-count)` to make a link

File: README.md [hunk 2]
```diff
@@ -57,9 +57,11 @@ fn hello(py: Python) -> PyResult<()> {
...
+Also on macOS, you will need to rename the output from \*.dylib to \*.so.
```
Comment (messense) [inline README.md]: Remove `Also`?

Comment (dkao1978) [reply README.md]: "Also" connects the two points. It wouldn't be the worst thing if it was removed.

**Comment by dkao1978** [comment]
Please let me know if any other changes you would like.

**Comment by fafhrd91** [comment]
@messense could you complete review?

File: guide/src/overview.md [hunk 3]
```diff
@@ -51,8 +51,8 @@ fn hello(py: Python) -> PyResult<()> {
 
 Example library with python bindings:
 
-The following two files will build with `cargo build`, and will generate a python-compatible library.
-On macOS, you will need to rename the output from \*.dylib to \*.so.
+The following two files will build with `cargo build`, and will generate a python-compatible library. For MacOS, "-C link-arg=-undefined -C link-arg=dynamic_lookup" is required to build the library. Setup.py includes this by default. See examples/word-count.
```
Comment (messense) [inline guide/src/overview.md]: I think it's better to have a link links to `https://github.com/PyO3/pyo3/tree/master/examples/word-count` here too, also in `lib.rs`

**Pull request merged**